src/langchain_google_cloud_sql_mssql/mssql_chat_message_history.py

```diff
@@ -0,0 +1,88 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import json
+from typing import List
+
+import sqlalchemy
+from langchain_core.chat_history import BaseChatMessageHistory
+from langchain_core.messages import BaseMessage, messages_from_dict
+
+from langchain_google_cloud_sql_mssql.mssql_engine import MSSQLEngine
+
+
+class MSSQLChatMessageHistory(BaseChatMessageHistory):
+    """Chat message history stored in a Cloud SQL MSSQL database."""
+
+    def __init__(
+        self,
+        engine: MSSQLEngine,
+        session_id: str,
+        table_name: str = "message_store",
+    ) -> None:
+        self.engine = engine
+        self.session_id = session_id
+        self.table_name = table_name
+        self._create_table_if_not_exists()
+
+    def _create_table_if_not_exists(self) -> None:
+        create_table_query = f"""IF NOT EXISTS (SELECT * FROM INFORMATION_SCHEMA.TABLES
+            WHERE TABLE_NAME = '{self.table_name}')
+        BEGIN
+        CREATE TABLE {self.table_name} (
+            id INT IDENTITY(1,1) PRIMARY KEY,
+            session_id NVARCHAR(MAX) NOT NULL,
+            data NVARCHAR(MAX) NOT NULL,
+            type NVARCHAR(MAX) NOT NULL
+        )
+        END;"""
+
+        with self.engine.connect() as conn:
+            conn.execute(sqlalchemy.text(create_table_query))
+            conn.commit()
+
+    @property
+    def messages(self) -> List[BaseMessage]:  # type: ignore
+        """Retrieve the messages from Cloud SQL"""
+        query = f'SELECT data, type FROM "{self.table_name}" WHERE session_id = :session_id ORDER BY id;'
+        with self.engine.connect() as conn:
+            results = conn.execute(
+                sqlalchemy.text(query), {"session_id": self.session_id}
+            ).fetchall()
+        # load SQLAlchemy row objects into dicts
+        items = [
+            {"data": json.loads(result[0]), "type": result[1]} for result in results
+        ]
```

Comment:
nit: I think you can one-line this
```suggestion
        items = [ {"data": json.loads(r[0]), "type": r[1]} for r in results ]
```

Reply:
I did have it as one line but this is how `black` is formatting it 🤷‍♂️

Reply:
I see you made results var smaller, will update